xarray/core/variable.py

```diff
@@ -761,7 +757,14 @@ def __getitem__(self, key) -> Self:
         array `x.values` directly.
         """
         dims, indexer, new_order = self._broadcast_indexes(key)
-        data = as_indexable(self._data)[indexer]
+        indexable = as_indexable(self._data)
+
+        if isinstance(indexer, BasicIndexer):
+            data = indexable[indexer]
+        elif isinstance(indexer, OuterIndexer):
+            data = indexable.oindex(indexer)
+        else:
+            data = indexable[indexer]
```

Comment:
These should be indexable properties btw:
https://numpy.org/neps/nep-0021-advanced-indexing.html#high-level-changes
```suggestion
        if isinstance(indexer, BasicIndexer):
            data = indexable[indexer]
        elif isinstance(indexer, OuterIndexer):
            data = indexable.oindex[indexer]
        else:
            data = indexable[indexer]
```

Reply:
Done!